lib/HLSL/DxilLinker.cpp

```diff
@@ -1063,7 +1063,10 @@ void DxilLinkJob::StripDeadDebugInfo(Module &M) {
         else
           SubprogramChange = true;
       } else {
-        SubprogramChange = true;
+        // Copy it in anyway even if there's no function. When function is inlined
+        // the function reference is gone, but the subprogram is still valid as
+        // scope.
+        LiveSubprograms.push_back(DISP);
```

Comment:
The only way to do that would be to look at all metadata that could possibly have a scope or a submodule, but it's really no big deal if a subprogram shows up that isn't part of the module. The biggest issue is subprograms not added to the CU, because newer LLVM will reject the entire debug info in that case.